core/amber/src/main/scala/edu/uci/ics/amber/engine/architecture/worker/promisehandlers/QueryStatisticsHandler.scala

```diff
@@ -16,6 +16,15 @@ trait QueryStatisticsHandler {
   this: WorkerAsyncRPCHandlerInitializer =>
 
   registerHandler { (msg: QueryStatistics, sender) =>
+    // report internal queue length if the gap > 30s
+    val now = System.currentTimeMillis()
+    if (now - lastReportTime > 30000) {
```

Comment:
can we define this `30000` as a static variable, or further more, make it configurable from file?

Reply:
I personally think 30 seconds is too large, most of workflows won't reach 30 seconds

Reply:
Let me make it a static variable.